nflow-engine/src/main/java/io/nflow/engine/service/WorkflowDefinitionService.java

```diff
@@ -78,42 +56,28 @@ public void setWorkflowDefinitions(@NFlow AbstractResource nflowNonSpringWorkflo
   }
 
   /**
-   * Add workflow definitions from the nflowNonSpringWorkflowsListing resource and persist
-   * all loaded workflow definitions.
-   * @throws IOException when workflow definitions can not be read from the resource.
-   * @throws ReflectiveOperationException when the workflow definition can not be instantiated.
+   * Persist all loaded workflow definitions if needed.
    */
-  public void postProcessWorkflowDefinitions() throws IOException, ReflectiveOperationException {
-    if (nonSpringWorkflowsListing == null) {
-      logger.info("No non-Spring workflow definitions");
-    } else {
-      initNonSpringWorkflowDefinitions();
-    }
+  public void postProcessWorkflowDefinitions() {
```

Comment:
Now that we store definitions in the addWorkflowDefinition already, we should just remove this `postProcessWorkflowDefinitions` method.
It will also nicely clean up some dependencies from other components.

Reply:
We can't remove it since we need to persist in different phase depending on autoinit, but I moved persisting logic completely to workflow definition service.